hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -245,7 +249,14 @@ public String extractPattern(TemporalUnit unit) {
 			case DAY_OF_YEAR:
 				return "({fn timestampdiff(sql_tsi_day, date(char(year(?2),4)||'-01-01'),?2)}+1)";
 			case DAY_OF_WEEK:
-				return "(mod({fn timestampdiff(sql_tsi_day, {d '2000-01-01'}, ?2)},7)+1)";
+				// Use the approach as outlined here: https://stackoverflow.com/questions/36357013/day-of-week-from-seconds-since-epoch
+				return "(mod(mod({fn timestampdiff(sql_tsi_day, {d '1970-01-01'}, ?2)}+4,7)+7,7)+1)";
```

Comment:
A test in `FunctionTests` failed because it produced the week number `21` instead of `22` I think.